aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -1118,4 +1119,41 @@ std::vector<Tensor> nonzero_numpy(const Tensor& self) {
   return self.nonzero().unbind(1);
 }
 
+Tensor & masked_scatter__cpu(Tensor& self, const Tensor & mask, const Tensor & source) {
+  at::assert_no_internal_overlap(self);
+  TORCH_CHECK(
+      self.scalar_type() == source.scalar_type(),
+      "masked_scatter: expected self and source to have same dtypes but got",
+      self.scalar_type(),
+      " and ",
+      source.scalar_type());
+
+  TORCH_CHECK(self.device().type() == at::kCPU, "device type of self (", self.device().type(), ") is not CPU");
+  TORCH_CHECK(mask.device().type() == at::kCPU, "device type of mask (", mask.device().type(), ") is not CPU");
+  TORCH_CHECK(source.device().type() == at::kCPU, "device type of source (", source.device().type(), ") is not CPU");
+
+  Tensor b_mask;
+  std::tie(b_mask) = expand_inplace(self, mask, "masked_scatter_");
+
+  if (b_mask.dtype() == ScalarType::Byte) {
+    TORCH_WARN("masked_scatter_ received a mask with dtype torch.uint8, this behavior is now deprecated," \
+            "please use a mask with dtype torch.bool instead.");
+  }
+
+  TORCH_CHECK(self.numel() == b_mask.numel(), "Number of elements of self != Number of elements in mask");
```

Comment:
why is this check needed? `expand_inplace` should have failed if it could not satisfy this invariant?

Reply:
Right. This is redundant. Thanks!